src/platform/Linux/bluez/Helper.cpp

```diff
@@ -1043,13 +1018,17 @@ static void bluezObjectsSetup(BluezEndpoint * apEndpoint)
         }
         g_list_free_full(interfaces, g_object_unref);
     }
+
     VerifyOrExit(apEndpoint->mpAdapter != nullptr, ChipLogError(DeviceLayer, "FAIL: NULL apEndpoint->mpAdapter in %s", __func__));
-    bluez_adapter1_set_powered(apEndpoint->mpAdapter, TRUE);
+
+    if (bluez_adapter1_get_powered(apEndpoint->mpAdapter) != TRUE)
+        bluez_adapter1_set_powered(apEndpoint->mpAdapter, TRUE);
 
     // Setting "Discoverable" to False on the adapter and to True on the advertisement convinces
     // Bluez to set "BR/EDR Not Supported" flag. Bluez doesn't provide API to do that explicitly
     // and the flag is necessary to force using LE transport.
-    bluez_adapter1_set_discoverable(apEndpoint->mpAdapter, FALSE);
+    if (bluez_adapter1_get_discoverable(apEndpoint->mpAdapter) != FALSE)
```

Comment:
Why this check? Something was wrong with simply setting the property to given value?